File: 2023/Arrays/ContainerWithMostWater.py
def maxArea(height):
    ws = 0
    we = len(height) - 1
    max_area = 0
    while ws < we:
        width = we - ws
        min_height = min(height[ws], height[we])
        area = width * min_height
        max_area = max(max_area, area)
        if height[ws] < height[we]:
            ws += 1
        else:
            we -= 1
    return max_area

File: 2023/Arrays/ContainerWithMostWater.cs
class ContainerWithMostWater {
    static int maxArea(int[] height) {
        int ws, we, width, minHeight, area, maxArea;
        ws = 0;
        we = height.Length - 1;
        maxArea = 0;
        while (ws < we) {
            width = we - ws;
            minHeight = Math.Min(height[ws], height[we]);
            area = width * minHeight;
            maxArea = Math.Max(maxArea, area);
            if (height[ws] < height[we]) {
                ws++;
            } else {
                we--;
            }
        }
        return maxArea;
    }
}